docusaurus-docs/community/policies/backwards-compatibility.md

```diff
@@ -139,42 +139,72 @@ get community feedback.
 
 #### Removing versions of API endpoints
 
-It is not recommended to remove versions of API endpoints. Removing API
-endpoints, or versions of endpoints, breaks backwards compatibility and should
-only be done under exceptional circumstances such as a security vulnerability.
+Removal of an API endpoint is sometimes necessary. Deprecation notices and removals will
+always be available in [release notes](https://github.com/conda-incubator/conda-store/blob/main/CHANGELOG.md). Further, documentation should be updated to reflect these changes. This should include:
+- version number of the release where this was deprecated
+- provide suggestions for alternatives (if possible)
+- provide justification for the removal (such as a link to the issue
+  or CVE that necessitated the removal).
+
+In order to make these kind of breaking changes responsibly, follow the steps
+outlined below.
+
+##### 1. Deprecate the endpoint
+
+In this stage, conda-store should not be introducing a breaking change. The
+deprecation step communicates to users that the endpoint is marked for removal
+and will removed in a future release.
 
-If the desire is to prevent a developer from relying on an API endpoint, adding
-a warning to the API documentation along with a recommended alternative should
-be used rather than a deprecation or removal.
+:::info[For vulnerabilities]
 
-In the case of a removed endpoint, or endpoint version, conda-store should
-return a status code of `410 Gone` to indicate the endpoint has been removed
-along with a json object stating when and why the endpoint was removed and what
+In the case of a serious security vulnerability, conda-store may skip the deprecation
+step and remove the endpoint immediately.
+
+:::
+
+Deprecations in the REST API follow the
+form outlined by the [deprecation header RFC](https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-deprecation-header-02). To deprecate an endpoint add the following
+response headers to the endpoint
+
+```
+{
+   "Deprecation": "True",
+   "Sunset": <removal date, eg. "Mon, 16 Feb 2025 23:59:59 UTC" >
+}
+```
+
+The "removal date" indicates the date after which conda store will no longer serve
+the endpoint. It will be specified as a [HTTP-Date](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date). The removal date can be set to (at earliest) the date of the target
+next release.
```

Comment:
See comment above:

> The removal date can be set to (at earliest) the date of the target
next release.

Since we use CalVer and _do not_ have set release dates, this puts us in a position where we need to agree on a release and meet it no matter what.
Since we do not have a set release schedule, this is rather ambiguous, so we need a clearer guideline here.

Reply:
I can be more specific here. 

I think sunset date should be set to about 2 months from the release that the deprecation is going out in. 
* Any time before this date, users should expect this endpoint to work
* Any time after this date, the endpoint may still be available (due to no release having gone out), but users should expect that this endpoint will disappear

For example, say you have an endpoint `/hello-planet` that is getting deprecated. The whole deprecation cycle will be something like:
* 2024/12/17 - add deprecation headers to `/hello-planet`,  sunset date is `2025/03/17`
* 2024/12/19 - new release goes out (sunset date is more than 2 months away, so everything is good)
* 2025/01/30 - new release goes out, `/hello-planet` is still deprecated
* 2025/03/16 - new emergency release goes out to fix a discovered security vulnerability. Still before the `/hello-planet` sunset date, so it remains in the deprecated state
* 2025/03/17 - `/hello-planet` sunset date. No release so this endpoint is still functional, however will be yanked at next release
* 2025/03/29 - new release goes out `/hello-planet` endpoint functionality is removed, returns a `410` response with removal date (eg. 2025/04/29)
* 2025/04/29 - new release goes out. Happens to coincide with the removal date. Can remove the endpoint as part of this release

Reply:
I don't think the removal of functionality step offers much more help to users than the deprecation step. If the team is open to changing this more, I would advocate for jumping straight from deprecating the endpoint to removing the endpoint after the sunset date.